bin/check-http-json.rb

```diff
@@ -188,20 +190,26 @@ def acquire_resource
       message = "key has expected value: '#{config[:key]}' "
       if config[:value]
         raise "unexpected value for key: '#{leaf}' != '#{config[:value]}'" unless leaf.to_s == config[:value].to_s
-        message += " equals '#{config[:value]}'"
+        message += "equals '#{config[:value]}'"
       end
       if config[:valueGt]
         raise "unexpected value for key: '#{leaf}' not > '#{config[:valueGt]}'" unless leaf.to_f > config[:valueGt].to_f
-        message += " greater than '#{config[:valueGt]}'"
+        message += "greater than '#{config[:valueGt]}'"
       end
       if config[:valueLt]
         raise "unexpected value for key: '#{leaf}' not < '#{config[:valueLt]}'" unless leaf.to_f < config[:valueLt].to_f
-        message += " less than '#{config[:valueLt]}'"
+        message += "less than '#{config[:valueLt]}'"
       end
 
       ok message
     rescue => e
-      critical "key check failed: #{e}"
+      if config[:dump_json]
+        json_response = config[:pretty] ? JSON.pretty_generate(json) : json
```

Comment:
This _should_ only dump the parsed json from the body of the response after it's been validated, and we're trying to match a specific key/value (https://github.com/jplindquist/sensu-plugins-http/blob/fec20b0bac993a02fabdeedbce67fd70ed6d7bc4/bin/check-http-json.rb#L181-L183).  It should not include any headers of the request or response at all as far as I know.  

I'll be happy to update it if you want with additional checks if auth headers are set, but otherwise this is just a straight dump of the parsed json body after the request is made and we've received a valid json response

Reply:
you are correct on closer inspection.